psclient/src/main/java/com/majeur/psclient/service/ReplayManager.kt

```diff
@@ -0,0 +1,110 @@
+package com.majeur.psclient.service
+
+import com.majeur.psclient.service.observer.BattleRoomMessageObserver
+import com.majeur.psclient.ui.BaseFragment
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.json.JSONObject
+
+class ReplayManager(
+        val httpClient: OkHttpClient,
+        val replayCallback: ReplayCallback) {
+
+    private val fragmentScope = BaseFragment.FragmentScope()
```

Comment:
FragmentScope is used to bind fragment's lifecycle to its background jobs to avoid such job from posting results to a destroyed or destroying fragment. Here, we are in the service scope so we do not have to use this.
You can use regulars callbacks from OkHttp (such as [here](https://github.com/MajeurAndroid/Android-Unofficial-Showdown-Client/blob/37febe67fd350c63e569f059d25e87432b58053c/psclient/src/main/java/com/majeur/psclient/service/ShowdownService.kt#L209))

Reply:
Thank you for the explanation! That makes a lot of sense.

I updated it to use the regular callbacks, but I had to add a Handler object just so I could post to the UI. (same way that ShowdownService does it)